### DISCUSSION TITLE
在BCLInux8.8编译单机版

### DISCUSSION THREAD

**Discussion by zs001122** [milvus-io/milvus#37669] [Q&A and General discussion]
![image](https://github.com/user-attachments/assets/7e58013e-f5d2-48d4-9fa9-91ef39d3f985)
我执行完./scripts/install_deps.sh后再执行make milvus,出现这个警告，我在根目录下的internal/core也没有找到对应的.so文件

**Comment by yhmo** [comment]
你读一下Makefile里的内容，当你make milvus的时候，它会做这三件事：
milvus: build-cpp print-build-info build-go

milvus_core.so是由build-cpp编出来的
```
build-cpp: generated-proto
	@echo "Building Milvus cpp library ..."
	@(env bash $(PWD)/scripts/core_build.sh -t ${mode} -a ${use_asan} -n ${use_disk_index} -y ${use_dynamic_simd} ${AZURE_OPTION} -x ${index_engine} -o ${use_opendal})
```

build-cpp实际上是执行scripts/core_build.sh这个脚本，估计是脚本没执行成功，至于为什么不成功，恐怕我们也帮不上忙了。
可以试一下在core_build.sh脚本的开始处加上这行，把azure排除在外，在有些用户的机器上就能编通
BUILD_WITHOUT_AZURE=ON

**Reply by zs001122** [reply]
我看到milvus 2.3.16有rpm包，它跟最新的在储存跟检索上有差距吗？差距大不大？😀
编译实在太难搞了，我在ubuntu上编译成功，但是在这个BCLinux上一直有问题

**Reply by yhmo** [reply]
2.3.16在性能上跟2.4差别不大

**Comment by xiaofan-luan** [comment]
直接用docker不行么？

**Reply by zs001122** [reply]
生产环境原因不能使用docker
